iree/hal/cuda/status_util.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/hal/cuda/status_util.h"
+
+#include "iree/hal/cuda/dynamic_symbols.h"
+
+iree_status_t iree_hal_cuda_result_to_status(
+    iree::hal::cuda::DynamicSymbols* syms, CUresult result, const char* file,
+    uint32_t line) {
+  if (result == CUDA_SUCCESS) {
```

Comment:
you could IREE_LIKELY() this --- or put the ok check as part of the macros you are using as well to allow that to be inlined more effectively (otherwise, every cuda call is likely to route through this function, as although LTO is pretty great it's not usually smart enough to do something like inline just this if and leave the rest in the called function)